### DISCUSSION TITLE
Associate `Set List` to the corresponding exercise in the `Exercise List`

### DISCUSSION THREAD

**Discussion by JamieBort** [wger-project/wger#902] [Q&A] [answered]
How to I find out which exercise from my `Set List` corresponds to the exercise in the `Exercise List`?

When I hit the `Workout Log List` endpoint (GET /api/v2/workoutlog/) I get the following result:
```
{
    "count": 14,
    "next": null,
    "previous": null,
    "results": [
        {
            "id": 3634134,
            "reps": 1,
            "weight": "0.00",
            "date": "2021-11-29",
            "rir": null,
            "exercise": 398,
            "workout": 305097,
            "repetition_unit": 1,
            "weight_unit": 6
        },
        .
        .
        .
   ]
}
```
Where `"exercise": 398,` corresponds to 

```
{
    "count": 419,
    "next": null,
    "previous": null,
    "results": [
        .
        .
        .
        {
            "id": 398,
            "uuid": "2359347e-0561-4d03-b050-b3c18f9d7b37",
            "name": "Run - Interval Training ",
            "exercise_base": 529,
            "status": "2",
            "description": "<p>Run and do some interval trainings such as hill repat, fartlek,..</p>",
            "creation_date": "2016-05-11",
            "category": 9,
            "muscles": [],
            "muscles_secondary": [],
            "equipment": [],
            "language": 2,
            "license": 2,
            "license_author": "BeLikeWater",
            "variations": []
        },
        .
        .
        .
   ]
}
```
from `Exercise List` (GET api/v2/exercise/?limit=420) endpoint.

When I hit the `Set List` endpoint (GET /api/v2/set/) I get this result:
```
{
    "count": 7,
    "next": null,
    "previous": null,
    "results": [
        {
            "id": 294672,
            "exerciseday": 155390,
            "sets": 3,
            "order": 1,
            "comment": "Could also add narrow or wide stance among others maybe."
        },
        .
        .
        .
   ]
}
```
`"id": 294672,` is auto-assigned.

For what it's worth, I've studied the diagram found [here](https://wger.de/static/images/api/db-all.svg). ([Source](https://wger.de/en/software/api))

Thank you very much in advance.

**Comment by rolandgeider** [comment, marked as answer]
you are right, there is no direct connection between logs and sets, only to workout. We should have an issue to add a nullable foreign key to the table, till then you will need to go over the workoutID / exerciseID and the date

**Reply by rolandgeider** [reply]
BTW you can directly access a specific object over the API with `/api/v2/exercise/398/`

**Reply by JamieBort** [reply]
> BTW you can directly access a specific object over the API with /api/v2/exercise/398/

Thank you for that detail.

**Comment by JamieBort** [comment]
Thank you for the helpful and quick response. 
Did you want me to create that issue?
I don't mind either way. 
I don't want to duplicate efforts but also know you're busy.

**Reply by rolandgeider** [reply]
Sure go ahead

**Comment by JamieBort** [comment]
Hi. If/when you have a moment I could use some guidance. I attempted to
join your discord server but all of the links to it seem to be broken.
I also don't mind asking elsewhere. But I am not sure where to turn other
than to you via email.

I just created this issue:
https://github.com/wger-project/wger/issues/911

I figured the best thing to do is to first get the app running locally. So
that I can figure out how it all works.

For example, I don't know which file(s) update the database nor which ones
make calls to it.

But I needed some guidance on running it locally. (I am much less
familiar with Python than JavaScript)

Do I need to run just a docker instance?
https://wger.readthedocs.io/en/latest/docker.html#

Or does it suffice to follow only the Development instructions?
https://wger.readthedocs.io/en/latest/development.html

Or do I need to run both?

At the moment I've followed just the Development instructions.
See the output from Terminal below:

(venv-wger) ***@***.***:
> *~/Documents/DevFiles/client_projects/wger/src*$ python manage.py
> runserver
>
> Watching for file changes with StatReloader
>
> Performing system checks...
>
>
> System check identified no issues (0 silenced).
>
> December 10, 2021 - 02:00:53
>
> Django version 3.2.10, using settings 'settings'
>
> Starting development server at http://127.0.0.1:8000/
>
> Quit the server with CONTROL-C.
>

My next step was going to be to run the appropriate script found in the
"src/package.json" file. So that I could interact with the app.
However I first navigated to "http://127.0.0.1:8000/" in my browser and
logged in with the admin credentials at the bottom of the Development
instructions documentation:

>
>    -
>
>    *username*: admin
>    -
>
>    *password*: adminadmin
>
> However I am not able to use the app. Please see the last three
screenshots.

Thank you very much.

Jamie

**Reply by rolandgeider** [reply]
Mhhh, can you try again to join the discord server? The link should definitely work (unless I've copy/pasted it wrong somewhere) https://discord.gg/rPWFv6W

You can either use a docker instance or install it "locally", whatever is best for you. I would suggest you to try to join the server and ping me there, that's usually faster but if it still doesn't work we'll solve it here
